source/jormungandr/jormungandr/street_network/here.py

```diff
@@ -97,17 +114,50 @@ def __init__(
         self.matrix_service_url = 'https://matrix.{base_url}/calculatematrix.json'.format(
             base_url=service_base_url
         )
-        self.api_id = api_id
-        self.api_code = api_code
+        self.log = logging.LoggerAdapter(logging.getLogger(__name__), extra={'streetnetwork_id': text_type(id)})
+        self.apiKey = apiKey
         self.modes = modes
         self.timeout = timeout
-        self.max_points = 100  # max number of point asked in the routing matrix
+        self.matrix_type = matrix_type
+        if (
+            self.matrix_type != MatrixType.simple_matrix.value
+            and self.matrix_type != MatrixType.multi_direct_path.value
+        ):
+            self.log.error(
```

Comment:
You could just simply write

```suggestion
        if matrix_type in MatrixType:
            self.matrix_type = matrix_type
        else:
            self.log.error(
```

Reply:
hooo sweet